backblaze/store.go

```diff
@@ -0,0 +1,458 @@
+		if err != nil {
+			return nil, err
+		}
+		return NewStore(client, conf)
+	})
+}
+
+type (
+	// FS Simple wrapper for accessing backblaze blob files, it doesn't currently implement a
+	// Reader/Writer interface so not useful for stream reading of large files yet.
+	FS struct {
+		ID        string
+		client    *bz.B2
+		bucket    string
+		cachepath string
+	}
+
+	blobRef struct {
+		fs         *FS
+		name       string
+		bucket     *bz.Bucket
+		updated    time.Time
+		opened     bool
+		cachedcopy *os.File
+		cachepath  string
+		rc         io.ReadCloser
+		readonly   bool
+		metadata   map[string]string
+	}
+)
+
+// NewClient create new AWS s3 Client.  Uses cloudstorage.Config to read
+// necessary config settings such as bucket, region, auth.
+func NewClient(conf *cloudstorage.Config) (*bz.B2, error) {
+
+	switch conf.AuthMethod {
+	case AuthKey:
+
+		account := conf.Settings.String(Account)
+		if account == "" {
+			return nil, ErrNoAccount
+		}
+		key := conf.Settings.String(Key)
+		if account == "" {
+			return nil, ErrNoAccessKey
+		}
+
+		b2, err := bz.NewB2(bz.Credentials{
+			AccountID:      account,
+			ApplicationKey: key,
+		})
+		if err != nil {
+			gou.Warnf("could not get backblaze client %v", err)
+			return nil, err
+		}
+
+		return b2, err
+	}
+
+	return nil, ErrNoAuth
+}
+
+// NewStore Create Backblaze client of type cloudstorage.Store
+func NewStore(c *bz.B2, conf *cloudstorage.Config) (*FS, error) {
+
+	if conf.TmpDir == "" {
+		return nil, fmt.Errorf("unable to create cachepath. config.tmpdir=%q", conf.TmpDir)
+	}
+
+	err := os.MkdirAll(conf.TmpDir, 0775)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create cachepath. config.tmpdir=%q err=%v", conf.TmpDir, err)
+	}
+
+	uid := uuid.NewUUID().String()
+	uid = strings.Replace(uid, "-", "", -1)
+
+	return &FS{
+		client:    c,
+		bucket:    conf.Bucket,
+		cachepath: conf.TmpDir,
+		ID:        uid,
+	}, nil
+}
+
+// Type is he Store Type [google, s3, azure, localfs, etc]
+func (f *FS) Type() string {
+	return StoreType
+}
+
+// Client gets access to the underlying native Client for Backblaze
+func (f *FS) Client() interface{} {
+	return f.client
+}
+
+// Get returns an object (file) from the cloud store. The object
+// isn't opened already, see Object.Open()
+// ObjectNotFound will be returned if the object is not found.
+func (f *FS) Get(ctx context.Context, o string) (cloudstorage.Object, error) {
+	return nil, ErrNotSupported
+}
+
+// Objects returns an object Iterator to allow paging through object
+// which keeps track of page cursors.  Query defines the specific set
+// of filters to apply to request.
+func (f *FS) Objects(ctx context.Context, q cloudstorage.Query) (cloudstorage.ObjectIterator, error) {
+	return nil, ErrNotImplemented
+}
+
+// List file/objects filter by given query.  This just wraps the object-iterator
+// returning full list of objects.
+func (f *FS) List(ctx context.Context, q cloudstorage.Query) (*cloudstorage.ObjectsResponse, error) {
```

Comment:
I am assuming these are not implemented because you didn't need them?   Ie, would be available for implementation at some point as opposed to their api doesn't support it?  Just curious.

Reply:
Hi Araddon, I am looking at https://godoc.org/gopkg.in/kothar/go-backblaze.v0 and I found that there's a list method: func (b *Bucket) ListFileNames(startFileName string, maxFileCount int) (*ListFilesResponse, error). Maybe I could implement your interface using it. But I think that context doesn't matter in this case.